### PR TITLE
pin pydantic < 2.9

### PR DESCRIPTION
**Pull request by mattseddon**
2.9.0 has breaking changes that are beyond me solving in the next 20 minutes

Failures: 

- https://github.com/iterative/datachain/actions/runs/10726012473
- https://github.com/iterative/datachain/actions/runs/10724648636

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -40,7 +40,7 @@ dependencies = [
   "dill==0.3.8",
   "cloudpickle",
   "orjson>=3.10.5",
-  "pydantic>=2,<3",
+  "pydantic==2.8.2",
```
Comment (dmpetrov) [inline pyproject.toml:43]: Shouldn't it be '>2,<2.9'?

**Review by dmpetrov — APPROVED** [review]
LG
See a comment

File: pyproject.toml (hunk 1 above)
Comment (mattseddon) [reply pyproject.toml:43]: I don't understand why we are so laissez-faire with dependency versions

Comment (rlamy) [reply pyproject.toml]: Exact pins are good for apps, but terrible for libraries.

**Pull request merged**